sitemap.functions.php

```diff
@@ -365,6 +371,11 @@ function scan_url($url)
         return $depth--;
     }
 
+    if ($noindex && (preg_match_all('/\<meta.*?\>/mis',$html,$ar) and strstr(join(',',$ar[0]),'noindex'))) {
+        logger("This page is set to noindex.", 1);
+        return $depth--;
```

Comment:
Instead of returning here, can you set a flag `$is_noindex_url` to skip `fwrite($file_stream, $map_row);` so that child links to be processed?